templ/dotenv.tmpl

```diff
@@ -0,0 +1,60 @@
+{{- define "item" }}
+  {{- $comma := false -}}
+# {{ .Doc }}
+  {{- if .EnvName -}}
+    {{ if eq .EnvSeparator "," }}
+      {{- $comma = true -}}
+# (comma-separated
+    {{ else if ne .EnvSeparator "" }}
+      {{- $comma = true -}}
+# (separated by `{{.EnvSeparator}}`
+    {{ end -}}
+    {{- if .Required -}}
+      {{- if $comma -}}, {{- else -}}
+# (
+{{- end -}}
+      {{- $comma = true -}}
+      required
+    {{- end -}}
+    {{- if .Expand -}}
+      {{- if $comma }}, {{ else }}# ({{ end -}}
+      {{- $comma = true -}}
+      expand
+    {{- end -}}
+    {{- if .NonEmpty -}}
+      {{- if $comma }}, {{ else }}# ({{ end -}}
+      {{- $comma = true -}}
+      non-empty
+    {{- end -}}
+    {{if .FromFile -}}
+      {{- if $comma }}, {{ else }}# ({{ end -}}
+      {{- $comma = true -}}
+      from-file
+    {{- end -}}
+    {{- if ne .EnvDefault "" -}}
+      {{- if $comma }}, {{ else }}# ({{ end -}}
+      {{- $comma = true -}}
+      {{- .EnvDefault | printf "default: `%s`" -}}
+    {{- end -}}
+    {{- if $comma }}){{ end -}}
+  {{- end }}
+  {{- $children := .Children 2 }}
+  {{- if $children }}
+    {{- range $children }}
+{{ template "item" . }}
+    {{- end }}
+  {{- end -}}
+
+    Save me: {{- .EnvName | printf "* `%s`" -}}
```

Comment:
I suppose this could be changed to:
```templ
  {{- end }}
{{ printf "%s=%s" .EnvName .EnvDefault -}}
```

to keep the newline at the beginning and using default value as a placeholder

Reply:
hmm ok, curious as to the reason behind using `printf` instead of just `{{.EnvName}}={{.EnvDefault}}`?

Reply:
Initially it was created to strip training newlines and spaces with dash at the end of the pipeline `-}}`

Reply:
oooh gotcha, ok, good to know